### PR TITLE
Implement create product using XML data

### PR DESCRIPTION
**Pull request by dplumb94**
This is the first PR to implement XML support for product. The code was getting a bit complex so I am submitting this one first to keep it incremental. Remaining tasks include:

- Implementing update
- Validation
- Fetching alternate ID to fill in the owner

To test: 

1. start up the `examples/splinter/docker-compose.yaml` network and create a Grid circuit between alpha and beta
2. Set up agents and organizations as follows:

```
grid keygen alpha-agent

export GRID_SERVICE_ID=<circuit ID>::gsAA

grid organization create crgl Cargill --metadata gs1_company_prefixes=0
grid role create crgl productowner --permissions schema::can-create-schema,product::can-create-product --active
grid agent update crgl $(cat ~/.grid/keys/alpha-agent.pub) --active --role productowner --role admin
```
3. submit a transaction to create a schema with the new `xsd_schema.yaml` file. If you are executing this from the gridd-alpha container you might need to copy it in. 
`$ grid schema create xsd-schema.yaml`
4. Create a product using some example XML data that conforms to the GridTradeItems XSD.
`$ grid product create -f yourproduct.yaml`

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -1946,6 +1956,23 @@ fn run() -> Result<(), CliError> {
                 .or_else(|| env::var(GRID_SERVICE_ID).ok());
 
             match m.subcommand() {
+                #[cfg(feature = "xml-support")]
+                ("create", Some(m)) if m.is_present("xml") => {
+                    let key = m
```
Comment (vaporos) [inline cli/src/main.rs:1961]: We should create a Grid CLI variant of https://github.com/Cargill/splinter/blob/main/cli/src/signing.rs so these key options work properly.

**Review by vaporos — CHANGES_REQUESTED** [review]

**Review by RyanLassigBanks — APPROVED** [review]
Tested locally

File: cli/src/main.rs [hunk 2]
```diff
@@ -766,8 +767,17 @@ fn run() -> Result<(), CliError> {
                                 .long("file")
                                 .short("f")
                                 .takes_value(true)
+                                .conflicts_with("xml")
                                 .help("Path to yaml file containing a list of products"),
                         )
+                        .arg(
+                            Arg::with_name("xml")
```
[the remark this answers is not in the document]
Comment (dplumb94) [reply cli/src/main.rs:774]: I think another option here would be to name the argument `--gdsn-3-1` (which could correspond with a more descriptive property name of the resulting Grid product definition). I think if we make it completely dynamic we could end up with some strange edge cases with duplicate GTIN product definitions being submitted etc. The complexity in solving that might not be proportional to the actual usefulness of the feature.

File: cli/src/main.rs (hunk 1 above)
Comment (dplumb94) [reply cli/src/main.rs:1974]: What isnt working properly? I can create a story for that

File: sdk/Cargo.toml [hunk 3]
```diff
@@ -89,7 +89,8 @@ experimental = [
     "sawtooth-compat",
     "sqlite",
     "track-and-trace",
-    "workflow"
+    "workflow",
+    "product-gdsn"
```
Comment (vaporos) [inline sdk/Cargo.toml:96]: This should be in alphabetical order after pike-rest-api

File: sdk/Cargo.toml [hunk 4]
```diff
@@ -114,3 +115,6 @@ sawtooth-compat = [
 ]
 sqlite = ["diesel/sqlite", "diesel_migrations"]
 workflow = []
+product-gdsn = [
```
Comment (vaporos) [inline sdk/Cargo.toml:121]: These should all be in alphabetical order as well (all features after experimental (experimental being special so separate)).

File: cli/src/main.rs [hunk 5]
```diff
@@ -776,8 +777,17 @@ fn run() -> Result<(), CliError> {
                                 .long("file")
                                 .short("f")
                                 .takes_value(true)
+                                .conflicts_with("xml")
                                 .help("Path to yaml file containing a list of products"),
                         )
+                        .arg(
```
Comment (vaporos) [inline cli/src/main.rs:783]: This should be behind a feature

File: cli/src/main.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (vaporos) [reply cli/src/main.rs:784]: "--gdsn-3-1" would work but its less user-friendly as it should be easy to determine which type of file it is at runtime, with just --file arguments.

File: cli/src/main.rs (hunk 1 above)
Comment (vaporos) [reply cli/src/main.rs:1974]: The first is the approach to looking up and handling the key argument, which should follow what is done in the splinter CLI, using code similar to that linked above; the second piece is creating a signer early and not passing the key into functions (pass the signer instead). splinter CLI has many examples of this.

**Review by vaporos — CHANGES_REQUESTED** [review]

File: cli/Cargo.toml [hunk 6]
```diff
@@ -70,9 +70,10 @@ experimental = [
     # The experimental feature extends stable:
     "stable",
     # The following features are experimental:
+    "product-gdsn",
     "purchase-order",
     "splinter",
-    "sqlite",
+    "sqlite"
```
Comment (vaporos) [inline cli/Cargo.toml:76]: Add the , back to this line pls

**Review by vaporos — APPROVED** [review]

**Review by dnewh — APPROVED** [review]

**Pull request merged**